cognify/optimizer/core/unified_layer_opt.py

```diff
@@ -385,7 +387,35 @@ def add_constraint(self, score, trial: optuna.trial.Trial):
             trial.set_user_attr(qc_identifier, constraint_result)
             # NOTE: add system attr at loading time
             # trial.set_system_attr(_base._CONSTRAINTS_KEY, constraint_result)
-
+    
+    def _update_best_trial(self, eval_result: EvaluationResult):
+        with self._study_lock:
+            current_score, current_cost = self.get_eval_feedback(eval_result)
+            if not self._should_stop and self._patience_budget is not None:
+                impv = False
+                score_threshold = self.top_down_info.opt_config._early_stop_quality_delta
+                cost_threshold = self.top_down_info.opt_config._early_stop_cost_delta
+                # reset if score or cost is improved
+                if current_score is not None and current_score >= score_threshold * (1 + score_threshold):
```

Comment:
why is this `score_threshold * (1 + score_threshold)`? shouldn't it be comparing to previous score?

Reply:
now using the best value so far to decide convergence.